File: SecondYear/semestrul4/MPP/PROIECT/JavaProjectORM/ProjectServer/src/main/java/ro/mpp2025/javaprojectui/service/ProbaService.java
package ro.mpp2025.javaprojectui.service;


import ro.mpp2025.javaprojectui.Proba;
import ro.mpp2025.javaprojectui.database.ProbaRepository;

import java.sql.SQLException;
import java.util.Optional;

public class ProbaService extends AbstractService<Integer, Proba>{
    private final ProbaRepository probaRepository;

    public ProbaService(ProbaRepository repository) {
        super(repository);
        this.probaRepository = repository;
    }

    public Proba findProbaByName(String name) {
        return probaRepository.findProbaByName(name);
    }

    public Proba findProbaByNameAndRange(String name, Integer range) {
        return probaRepository.findProbaByNameAndRange(name, range);
    }

    public Optional<Proba> update(Proba p) throws SQLException {
        return probaRepository.update(p);
    }

    public void delete(Integer id) throws SQLException {
        probaRepository.delete(id);
    }
}

File: SecondYear/semestrul4/MPP/PROIECT/CSharpProjectNetworking/Server/service/ProbaService.cs

using Model;
using Persistence.database;

namespace Server;
public class ProbaService : AbstractService<int, Proba>
{
    private readonly ProbaRepository probaRepository;

    public ProbaService(ProbaRepository repository) : base(repository)
    {
        this.probaRepository = repository;
    }

    public Proba FindProbaByName(string name)
    {
        return probaRepository.FindProbaByName(name);
    }

    public Proba FindProbaByNameAndRange(string name, int range)
    {
        return probaRepository.FindProbaByNameAndRange(name, range);
    }
}